From: Marcin Owsiany <marcin@owsiany.pl>
Subject: [PATCH] debian/enable-xpm

Enable XPM support.

Signed-off-by: Marcin Owsiany <marcin@owsiany.pl>

---
 Imakefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Imakefile b/Imakefile
index 539905c..69351be 100644
--- a/Imakefile
+++ b/Imakefile
@@ -22,7 +22,7 @@ XCOMM #define HAVE_HTML
 
 XCOMM If you wish to use colour xpm pixmaps for the icon and have the
 XCOMM Xpm library installed, then uncomment the following line: 
-XCOMM #define HAVE_XPM
+#define HAVE_XPM
 
 XCOMM If you wish to build with Motif, then uncomment the following line:
 XCOMM #define MOTIF
-- 
tg: (d3dafc5..) debian/enable-xpm (depends on: upstream)
